authLdap.php

```diff
@@ -254,10 +261,17 @@ function authLdap_login($user, $username, $password, $already_md5 = false)
             return false;
         }
 
-        $role = '';
+        // if the user exists, use that role
+        $uid = authLdap_get_uid($username);
+        $role = authLdap_user_role($uid);
+
         // do some group mapping
         if ($authLDAPGroupEnable) {
```

Comment:
Hmm, I know what you mean I think. With this implementation if the user has no role in LDAP anymore, the Wordpress role would stick.
So if we only read the user role from wordpress if either LDAP groups are disabled or if the wordpress role is more important than LDAP groups (```!$authLDAPGroupOverUser```), we should be fine (and potentially save a DB lookup).